traversal/procedure/ProcedureVertex.java

```diff
@@ -106,18 +101,47 @@ public ProcedureVertex.Type asType() {
         throw TypeDBException.of(ILLEGAL_CAST, className(this.getClass()), className(ProcedureVertex.Type.class));
     }
 
+    public int order() {
+        return order;
+    }
+
+    public List<ProcedureEdge<?, ?>> orderedOuts() {
+        if (orderedOuts == null) {
+            orderedOuts = outs().stream().sorted(Comparator.comparingInt(e -> e.to().order())).collect(toList());
+        }
+        return orderedOuts;
+    }
+
+    public Set<ProcedureEdge<?, ?>> transitiveOuts() {
+        if (transitiveOuts == null) {
+            HashSet<ProcedureEdge<?, ?>> transitive = new HashSet<>();
+            outs().forEach(edge -> {
+                transitive.add(edge);
+                transitive.addAll(edge.to().transitiveOuts());
+            });
+            transitiveOuts = transitive;
+        }
+        return transitiveOuts;
+    }
+
+    void setOrder(int order) {
+        this.order = order;
+    }
+
     @Override
     public String toString() {
-        String str = super.toString();
-        if (isStartingVertex) str += " (start)";
+        String str = order() + ": " + super.toString();
```

Comment:
`ProcedureVertex` prints its own order number - useful for debugging as well when inspecting variables!